spec/lib/contextual_logger_spec.rb

```diff
@@ -379,4 +379,45 @@ def expect_log_line_to_be_written(log_line)
       end
     end
   end
+
+  describe 'with redaction' do
+    let(:sensitive_data) { 'sensitive_string_123' }
+
+    before(:each) do
+      logger.redactor.match(sensitive_data)
```

Comment:
Ok, I see here that it's the actual sensitive value that has to be passed in, rather than its key. That seems pretty inflexible... and maybe just a bit insecure! (To redact credentials, I have to pass all the credentials in.)

Reply:
It's not insecure, because it's not storing it outside of memory, just like that rest of the application.  The only sure fire way to make sure that none of your secrets slip through into the logs, is to know what they actually are and fully redact them from the end log line.  Redaction based on context keys isn't because secrets aren't passed into logs accidentally just through context, they're also hidden in the message itself, as well as complex keys within the context that hold more information that just the secret.  You don't want to redact all information from a key just because a secret exists somewhere inside it.

The type of redaction you're thinking of is the way that Rails does it, which is request parameter passed, which won't work for this because of the use cases of not always having control over the code that is doing the logging, even though it might be logging a message that contains sensitive information.

This is not meant to be super flexible to start, it's meant to solve a current problem that exists, which is that secrets make their way into logs, regardless of the key.

Reply:
Yes, I was aware that they were still stored in memory. It did seem less secure to make another copy of it here, simply because that was another place they could leak from. But I do see your point that if you want to catch secrets that have leaked into text, that's the only way to do it.